codice/ddf
# DDF-2188 Adds new service definition for extracting metadata from text content files.

#### What does this PR do?
This PR adds a new service interface for reading the content of a text document and parsing out any specific metadata it may have to be added to its associated `Metacard`s. The two text importers - 
Tika and PDF - have been updated to find the `ContentMetadataExtractor` implementors and process each of them in turn.

This change also updates the `MetacardAttributeSecurityPolicyPlugin` so that it takes key-value pairs of attribute names. The metacard is checked to see if it has attributes named by those keys and then creates security policy map entries keyed by the values.

#### Who is reviewing it (please choose AT LEAST two reviewers that need to approve the PR before it can get merged; if a component team is listed, at least one of its members needs to approve)?
@rzwiefel @bcwaters @brendan-hofmann 

#### Choose 2 committers to review/merge the PR (please choose ONLY two committers from below, delete the rest).
@kcwire
@stustison

#### How should this be tested?
A full build/test run should be sufficient at this time. (An) implementation(s) of the `ContentMetadataExtractor` could be built and deployed and then tested against input files. ITests may be added before or after this is merged.

#### Any background context you want to provide?
This is a general solution to pulling extra metadata from the content of a text file. A specific use case would be reading banner or portion markings from a classified document and applying those to the associated `Metacard`.

#### What are the relevant tickets?
DDF-2188

#### Screenshots (if appropriate)
N/A

#### Checklist:
